client/fingerprint/secrets.go

```diff
@@ -21,13 +27,70 @@ func NewPluginsSecretsFingerprint(logger hclog.Logger) Fingerprint {
 
 func (s *SecretsPluginFingerprint) Fingerprint(request *FingerprintRequest, response *FingerprintResponse) error {
 	// Add builtin secrets providers
-	response.AddAttribute("plugins.secrets.nomad.version", "1.0.0")
-	response.AddAttribute("plugins.secrets.vault.version", "1.0.0")
+	defer func() {
+		response.AddAttribute("plugins.secrets.nomad.version", "1.0.0")
+		response.AddAttribute("plugins.secrets.vault.version", "1.0.0")
+	}()
 	response.Detected = true
 
+	pluginDir := request.Config.CommonPluginsDir
+	if pluginDir == "" {
+		return nil
+	}
+
+	secretsDir := filepath.Join(pluginDir, commonplugins.SECRETS_DIR)
+
+	files, err := helper.FindExecutableFiles(secretsDir)
+	if err != nil {
+		if os.IsNotExist(err) {
+			s.logger.Debug("secrets plugin dir does not exist", "dir", secretsDir)
```

Comment:
```suggestion
			s.logger.Trace("secrets plugin dir does not exist", "dir", secretsDir)
```

Seems like a lot of users will never bother.